Base/splitter.h

```diff
@@ -0,0 +1,42 @@
+#ifndef SPLITTER_H__
+#define SPLITTER_H__
+
+struct SplitterRecord
```

Comment:
How will Splitter communicate with the tree? The tree must receive from the splitter some info regarding the chosen split point. How do you suggest that communication? I thought of using the SplitterRecord as scikit.